api/errors/codes/codes.go

```diff
@@ -0,0 +1,331 @@
+	//      the files are deleted from the directory.
+	//
+	// HTTP Mapping: 400 Bad Request
+	// Google API Mapping: Code.FAILED_PRECONDITION
+	FailedPrecondition Code = 9
+
+	// Aborted means the operation was aborted, typically due to a concurrency issue
+	// such as a sequencer check failure or transaction abort.
+	//
+	// See the guidelines above for deciding between `FailedPrecondition`,
+	// `Aborted`, and `Unavailable`.
+	//
+	// HTTP Mapping: 409 Conflict
+	// Google API Mapping: Code.ABORTED
+	Aborted Code = 10
+
+	// OutOfRange means the operation was attempted past the valid range.
+	// E.g., seeking or reading past end-of-file.
+	//
+	// Unlike `InvalidArgument`, this error indicates a problem that may
+	// be fixed if the system state changes. For example, a 32-bit file
+	// system will generate `InvalidArgument` if asked to read at an
+	// offset that is not in the range [0,2^32-1], but it will generate
+	// `OutOfRange` if asked to read from an offset past the current
+	// file size.
+	//
+	// There is a fair bit of overlap between `FailedPrecondition` and
+	// `OutOfRange`.  We recommend using `OutOfRange` (the more specific
+	// error) when it applies so that callers who are iterating through
+	// a space can easily look for an `OutOfRange` error to detect when
+	// they are done.
+	//
+	// HTTP Mapping: 400 Bad Request
+	// Google API Mapping: Code.OUT_OF_RANGE
+	OutOfRange Code = 11
+
+	// Unimplemented means the operation is not implemented or is not
+	// supported/enabled in this service.
+	//
+	// HTTP Mapping: 501 Not Implemented
+	// Google API Mapping: Code.UNIMPLEMENTED
+	Unimplemented Code = 12
+
+	// Internal means an internal error. This means that some invariants expected
+	// by the underlying system have been broken. This error code is reserved
+	// for serious errors.
+	//
+	// HTTP Mapping: 500 Internal Server Error
+	// Google API Mapping: Code.INTERNAL
+	Internal Code = 13
+
+	// Unavailable means the service is currently unavailable. This is most likely a
+	// transient condition, which can be corrected by retrying with
+	// a backoff.
+	//
+	// See the guidelines above for deciding between `FailedPrecondition`,
+	// `Aborted`, and `Unavailable`.
+	//
+	// HTTP Mapping: 503 Service Unavailable
+	// Google API Mapping: Code.UNAVAILABLE
+	Unavailable Code = 14
+
+	// DataLoss means unrecoverable data loss or corruption.
+	//
+	// HTTP Mapping: 500 Internal Server Error
+	// Google API Mapping: Code.DATA_LOSS
+	DataLoss Code = 15
+
+	// Unauthenticated means the request does not have valid authentication
+	// credentials for the operation.
+	//
+	// HTTP Mapping: 401 Unauthorized
+	// Google API Mapping: Code.UNAUTHENTICATED
+	Unauthenticated Code = 16
+
+	// Application means there was an application error. This will typically be
+	// accompianed by a user-defined error name across the wire.
+	//
+	// HTTP Mapping: 500 Internal Server Error
+	// Google API Mapping: Code.UNKNOWN
+	Application Code = 17
```

Comment:
This is representing a User-Defined type per the doc I wrote. Note that application errors are errors, this is part of the idea of what we're doing here.